### DISCUSSION TITLE
Ctrl + [ mapping is not working with NvChad

### DISCUSSION THREAD

**Discussion by xander412** [NvChad/NvChad#2949] [Q&A] [answered]
This issue is a follow up of [NEOVIM - issue thread](https://github.com/neovim/neovim/discussions/29889) .

When I am mapping `vim.keymap.set('n', '<C-[>', '<C-t>', {silent = true, noremap = true})` in init.lua, the binding is not working. But when I use neovim --clean and set the mapping in normal mode, it is working. 

Is NvChad have a strict mapping of this key combination? How can I get rid of this ? I even tried delete the existing mapping of Ctrl + [ , but it was saying that there is not mapping. Please help me to bind this correctly.

**Comment by siduck** [comment]
![image](https://github.com/user-attachments/assets/9c3e2f1d-d011-4788-b988-d9a235f11134)
just tried your code and works

**Reply by xander412** [reply]
Were you able to navigate using ctags through your code ? forward and backward with Ctrl + [ and Ctrl + ] ?

Even for me: 
```
n  <C-[>       * <C-T>                                                                                                                                                                                
        Last set from Lua
```

The mapping is successful

**Reply by siduck** [reply]
idk what are ctags, try this mapping on other nvim configs, use nvimappname to test

**Reply by xander412** [reply]
I think I almost found the reason. @siduck , please let me know if the following binding is done in nvChad ?

```
n  <Esc>       * <Cmd> noh <CR>                                                                                                                                                                        
                 Clear highlights
        Last set from Lua

```

**Reply by xander412** [reply, marked as answer]
Solution : 
`lua/core/mappings.lua : 19`  NvChad : v2.0 
Comment the <Esc> binding. 

<C - [> was being interpreted as `<Esc>` by the terminal emulator causing this behaviour.

**Reply by siduck** [reply]
@xander412  nvchad v2.5 has no such mapping btw

**Reply by xander412** [reply]
Cool then. I don't know why I pulled the older version. Sorry for the inconvenience. I will stash my changes and pop them on the latest version. One question : Will this reinstall all the other modules especially mason modules ? Because I changed many things in pyright, cland modules which were installed in share/bin folder.

**Reply by siduck** [reply]
rm -rf .local/share/nvim will reinstall mason stuff too, so i advise back it up somewhere.

Also use the starter repo!
